lib/Controller/SignalingController.php

```diff
@@ -519,37 +521,66 @@ private function backendAuth(array $auth): DataResponse {
 	}
 
 	private function backendRoom(array $roomRequest): DataResponse {
-		$roomId = $roomRequest['roomid'];
+		$token = $roomRequest['roomid']; // It's actually the room token
 		$userId = $roomRequest['userid'];
 		$sessionId = $roomRequest['sessionid'];
 		$action = !empty($roomRequest['action']) ? $roomRequest['action'] : 'join';
-
-		try {
-			$room = $this->manager->getRoomByToken($roomId, $userId);
-		} catch (RoomNotFoundException $e) {
-			return new DataResponse([
-				'type' => 'error',
-				'error' => [
-					'code' => 'no_such_room',
-					'message' => 'The user is not invited to this room.',
-				],
-			]);
-		}
+		$actorId = $roomRequest['actorid'] ?? null;
+		$actorType = $roomRequest['actortype'] ?? null;
+		$inCall = $roomRequest['incall'] ?? null;
 
 		$participant = null;
-		if ($sessionId) {
+		if ($actorId !== null && $actorType !== null) {
 			try {
-				$participant = $room->getParticipantBySession($sessionId);
-			} catch (ParticipantNotFoundException $e) {
+				$room = $this->manager->getRoomByActor($token, $actorType, $actorId);
+			} catch (RoomNotFoundException $e) {
+				return new DataResponse([
+					'type' => 'error',
+					'error' => [
+						'code' => 'no_such_room',
+						'message' => 'The user is not invited to this room.',
+					],
+				]);
 			}
-		}
 
-		if (!empty($userId)) {
-			// User trying to join room.
 			try {
-				$participant = $room->getParticipant($userId);
+				if ($sessionId) {
+					try {
+						$participant = $room->getParticipantBySession($sessionId);
+					} catch (ParticipantNotFoundException $e) {
+						$participant = $room->getParticipantByActor($actorType, $actorId);
+					}
+				} else {
+					$participant = $room->getParticipantByActor($actorType, $actorId);
+				}
```

Comment:
If `getParticipantBySession` is called with [an empty `sessionId` it will also throw a `ParticipantNotFoundException`](https://github.com/nextcloud/spreed/blob/6996121c23afc68850d4e83c8f7fdae587dccecb/lib/Room.php#L380-L383). Maybe simplify the block removing the external if/else?

Reply:
When you provide a wrong session id, we actually shouldn't fallback, but just error out. like in the block when no actor id+type was given.
Fixed that now.